android/expoview/src/main/java/versioned/host/exp/exponent/modules/api/notifications/NotificationsModule.java

```diff
@@ -356,14 +365,121 @@ public void cancelScheduledNotificationAsync(final int notificationId, final Pro
     }
   }
 
+  @ReactMethod
+  public void cancelScheduledNotificationWithStringIdAsync(final String id, final Promise promise) {
+    try {
+      SchedulersManagerProxy.getInstance(getReactApplicationContext()
+          .getApplicationContext())
+          .removeScheduler(id);
+      promise.resolve(null);
+    } catch (Exception e) {
+      promise.reject(e);
+    }
+  }
+
   @ReactMethod
   public void cancelAllScheduledNotificationsAsync(final Promise promise) {
     try {
       ExponentNotificationManager manager = new ExponentNotificationManager(getReactApplicationContext());
       manager.cancelAllScheduled(mManifest.getString(ExponentManifest.MANIFEST_ID_KEY));
+
+      String experienceId = mManifest.optString(ExponentManifest.MANIFEST_ID_KEY, null);
+
+      SchedulersManagerProxy
+          .getInstance(getReactApplicationContext().getApplicationContext())
+          .removeAll(experienceId);
+
       promise.resolve(null);
     } catch (Exception e) {
       promise.reject(e);
     }
   }
+
+  @ReactMethod
+  public void scheduleNotificationWithTimer(final ReadableMap data, final ReadableMap optionsMap, final Promise promise) {
+    HashMap<String, Object> options = optionsMap.toHashMap();
+    int notificationId = Math.abs( new Random().nextInt() );
+    HashMap<String, Object> hashMap = data.toHashMap();
+    if (data.hasKey("categoryId")) {
+      hashMap.put("categoryId", getScopedIdIfNotDetached(data.getString("categoryId")));
+    }
+    HashMap<String, Object> details = new HashMap<>();
+    details.put("data", hashMap);
+    String experienceId;
+
+    try {
+      experienceId = mManifest.getString(ExponentManifest.MANIFEST_ID_KEY);
+      details.put("experienceId", experienceId);
+    } catch (Exception e) {
+      promise.reject(new Exception("Requires Experience Id"));
+      return;
+    }
+
+    IntervalSchedulerModel intervalSchedulerModel = new IntervalSchedulerModel();
```

Comment:
Same as with CalendarScheduler. Especially due to fact, that there is some code duplication in here.